packages-next/fields/src/types/file/Implementation.ts

```diff
@@ -1,164 +1,94 @@
-import { LocalFileAdapter } from './local-file';
-
-type StoredFile = {
-  id: string;
-  filename: string;
-  originalFilename: string;
-  mimetype: any;
-  encoding: any;
-  _meta?: Record<string, any>;
-};
-
-export class File<P extends string> extends Implementation<P> {
-  graphQLOutputType: string;
-  fileAdapter: LocalFileAdapter;
+import { getFileRef } from '@keystone-next/utils-legacy';
+import { FileData, KeystoneContext, BaseKeystoneList } from '@keystone-next/types';
+import { Implementation } from '../../Implementation';
+import { handleFileData } from './handle-file-input';
 
-  constructor(
-    path: P,
-    { adapter, ...configArgs }: FieldConfigArgs & { adapter: LocalFileAdapter },
-    extraArgs: FieldExtraArgs
-  ) {
-    super(path, { adapter, ...configArgs }, extraArgs);
-    this.graphQLOutputType = 'File';
-    this.fileAdapter = adapter;
-
-    if (!this.fileAdapter) {
-      throw new Error(`No file adapter provided for File field.`);
-    }
-  }
+export class FileImplementation<P extends string> extends Implementation<P> {
   get _supportsUnique() {
     return false;
   }
 
   gqlOutputFields() {
-    return [`${this.path}: ${this.graphQLOutputType}`];
-  }
-  gqlQueryInputFields() {
-    return [...this.equalityInputFields('String'), ...this.inInputFields('String')];
-  }
-  getFileUploadType() {
-    return 'Upload';
+    return [`${this.path}: FileFieldOutput`];
   }
+
   getGqlAuxTypes() {
     return [
-      `
-      type ${this.graphQLOutputType} {
-        id: ID
-        path: String
-        filename: String
-        originalFilename: String
-        mimetype: String
-        encoding: String
-        publicUrl: String
+      `enum FileMode {
+        local
       }
-    `,
+      input FileFieldInput {
+        upload: Upload
+        ref: String
+      }
+      type FileFieldOutput {
+        mode: FileMode!
+        filename: String!
+        filesize: Int!
+        ref: String!
+        src: String!
+      }`,
     ];
   }
-  // Called on `User.avatar` for example
-  gqlOutputFieldResolvers() {
-    return {
-      [this.path]: (item: Record<P, any>) => {
-        let itemValues = item[this.path];
-        if (!itemValues) {
-          return null;
-        }
-        if (this.adapter.listAdapter.parentAdapter.provider === 'sqlite') {
-          // we store document data as a string on sqlite because Prisma doesn't support Json on sqlite
-          // https://github.com/prisma/prisma/issues/3786
-          try {
-            itemValues = JSON.parse(itemValues);
-          } catch (err) {}
-        }
 
-        return {
-          publicUrl: this.fileAdapter.publicUrl(itemValues),
-          ...itemValues,
-        };
+  gqlAuxFieldResolvers() {
+    return {
+      FileFieldOutput: {
+        src(data: FileData, _args: any, context: KeystoneContext) {
+          if (!context.files) {
+            throw new Error('File context is undefined');
```

Comment:
Is this the error message we expect to see if we forget to include `config.files`? If so, we should probably make it more explicit about how to fix it.

Reply:
Replaced with "File context is undefined, this most likely means that you havent configurd keystone with a file config, see https://next.keystonejs.com/apis/config#files for details"